logic/resolvable/Concludable.java

```diff
@@ -53,16 +55,24 @@
 
     private final Map<Rule, Set<Unifier>> applicableRules;
     private final CONSTRAINT constraint;
+    private final Set<Constraint> coreConstraints;
 
-    private Concludable(CONSTRAINT constraint) {
+    private Concludable(CONSTRAINT constraint, Set<Constraint> otherCoreConstraints) {
```

Comment:
So this will get outdated quickly right? It will just take a Conjunction in the next iteration?

Reply:
Yes that's the plan!